pandas/core/groupby/generic.py

```diff
@@ -1759,7 +1761,39 @@ def _normalize_keyword_aggregation(kwargs):
         else:
             aggspec[column] = [aggfunc]
         order.append((column, com.get_callable_name(aggfunc) or aggfunc))
-    return aggspec, columns, order
+
+    # uniquify aggfunc name if duplicated in order list
+    uniquified_order = _uniquify_aggfunc(order)
+
+    # GH 25719, due to aggspec will change the order of assigned columns in aggregation
+    # uniquified_aggspec will store uniquified order list and will compare it with order
+    # based on index
+    aggspec_order = [
+        (column, com.get_callable_name(aggfunc) or aggfunc)
+        for column, aggfuncs in aggspec.items()
+        for aggfunc in aggfuncs
+    ]
+    uniquified_aggspec = _uniquify_aggfunc(aggspec_order)
+
+    # get the new indice of columns by comparison
+    col_idx_order = [uniquified_aggspec.index(o) for o in uniquified_order]
```

Comment:
Is it worth it (performance-wise) to convert this to an Index and do `.get_indexer()` to look up the indices?

Reply:
nice catch! not sure how much difference in performance will be, since this is only reordering columns index. but of course I could change it to `Index` and use `get_indexer()`, should be faster in theory.